lib/src/app_auth.dart

```diff
@@ -108,17 +108,20 @@ class AppAuth {
 
       return tokens["accessToken"];
     } on PlatformException catch (e) {
-      if (e.message?.toLowerCase()?.contains("network") == true) {
-        throw e.message; // probably bad network
-      }
```

Comment:
Shouldn't we keep this heuristic? Could map to another "nice" user facing message

Reply:
To check the occurrence of a single word in a description text... is is named a 'heuristic' - could call it a 'guess'.- 
My proposal tries to use all known exceptions to use the codes: 'timeout'..'failed_...',  Here we have a chance to simplify messages without considering every possible network error. 

I tested it with simulated timeouts, 7.x pods, the domain 'example.com' and absolute nonsense hostnames - as a user the visible messages are OK for me. (And internal logs are more detailed).  You have the last word on this ;-)

Reply:
Well, summarizing everything under two user facing messages, one for timeouts and one for everything else, makes it pretty impossible to even begin to troubleshoot the error.

When there's bad network conditions, like the network dropped during a request, a hostname could not be resolved etc, this tells us nothing about the state of API support at the user's pod. OTOH a bad network condition is nothing they need to bother their podmin about.

So improving their network connection is likely something the user can do something about, upgrading their pod likely is something they can't do anything about. It makes a lot of sense to me to try to separate these error classes, it will actually benefit the user a lot.